plugins/uefi-capsule/fu-uefi-bootmgr.c

```diff
@@ -292,6 +293,165 @@ fu_uefi_copy_asset (const gchar *source, const gchar *target, GError **error)
 	return TRUE;
 }
 
+static gboolean
+fu_uefi_bootmgr_setup_grub (FuDevice *device,
+			    const gchar *esp_path,
+			    const gchar *target_app,
+			    GError **error)
+{
+	g_autofree gchar *grub_reboot = NULL;
+	g_autofree gchar *grub_target = NULL;
+	g_autofree gchar *sysconfdir = NULL;
+
+	/* verify GRUB snippet exists */
+	sysconfdir = fu_common_get_path (FU_PATH_KIND_SYSCONFDIR);
+	grub_target = g_build_filename (sysconfdir, "grub.d", "35_fwupd", NULL);
```

Comment:
I wonder if generating a `/etc/grub.d/35_fwupd` snippet at runtime is the best approach for this.

What if this is instead a static file that's just shipped by `fwupd` but that's just a no-op if `EnableGrubChainLoad` isn't enabled in `/etc/fwupd/uefi_capsule.conf`? After all, the GRUB snippets are just shell scripts so there's no point on replacing strings IMO (i.e: could just do a `chainloader $EFI_PATH`).

Something like [30_uefi-firmware](https://git.savannah.gnu.org/cgit/grub.git/tree/util/grub.d/30_uefi-firmware.in):

```
EFI_VARS_DIR=/sys/firmware/efi/efivars
EFI_GLOBAL_VARIABLE=8be4df61-93ca-11d2-aa0d-00e098032b8c
OS_INDICATIONS="$EFI_VARS_DIR/OsIndicationsSupported-$EFI_GLOBAL_VARIABLE"

if [ -e "$OS_INDICATIONS" ] && \
   [ "$(( $(printf 0x%x \'"$(cat $OS_INDICATIONS | cut -b5)"\') & 1 ))" = 1 ]; then
  LABEL="UEFI Firmware Settings"

  gettext_printf "Adding boot menu entry for UEFI Firmware Settings ...\n" >&2

  cat << EOF
menuentry '$LABEL' \$menuentry_id_option 'uefi-firmware' {
	fwsetup
}
EOF
fi
```

Another option is to make the `fwupd` snippet even more static and just `source` from it a variable that's contained in another file (i.e: `/boot/grub2/fwupd.cfg`). That's what we do in Fedora to [support ACL](https://github.com/rhboot/grub2/blob/fedora-35/util/grub.d/01_users.in) of boot entries for example:

```
cat << EOF
if [ -f \${prefix}/user.cfg ]; then
  source \${prefix}/user.cfg
  if [ -n "\${GRUB2_PASSWORD}" ]; then
    set superusers="root"
    export superusers
    password_pbkdf2 root \${GRUB2_PASSWORD}
  fi
fi
EOF
```

Reply:
> Another option is to make the fwupd snippet even more static and just source from it a variable that's contained in another file (i.e: /boot/grub2/fwupd.cfg). 

I like this idea a lot.  When I get a chance I'll update the PR for it (unless @hughsie beats me to it).  Then we can also support systems that have read-only rootfs.  The fwupd package can just start shipping `/etc/grub.d/35_fwupd` and source the settings from somewhere actually writable.

>That's what we do in Fedora to support ACL of boot entries for example:

BTW TBH this sounds like it makes sense generally in GRUB, not just in Fedora.  Yall tried to upstream it?

Reply:
> unless @hughsie beats me to it

/me is neck deep in Redfish this week...

Reply:
> BTW TBH this sounds like it makes sense generally in GRUB, not just in Fedora. Yall tried to upstream it?

Yes, it's in our backlog for upstreaming (along with the [grub-set-password](https://github.com/rhboot/grub2/blob/fedora-35/util/grub-set-password.8) user-space tool we ship) but it's just that wasn't prioritized over the other downstream patches we are still carrying.

Reply:
I added something to match what you suggested (untested though so far)